File: Práctica 1/S1/src/s1/Proyecto.java
package s1;
public class Proyecto {
    
    private static Cliente cliente = new Cliente();
    
    public static void correrCarreras(){
        for(int i=0;i<cliente.getCarreras().size();i++){
            cliente.getCarreras().get(i).start();
        }
        try {
            Thread.sleep(100);
         } catch (Exception e) {
            System.out.println(e);
         }
    }
    
    public static void obtenerRankings(){
        for(int i=0;i<cliente.getCarreras().size();i++){
            cliente.getCarreras().get(i).mostrarRanking();
        }
    }
    
    public static void main(String[] args) {
        cliente.gestionar();
        cliente.getCarreras().get(0).mostrarBicicletas();
        correrCarreras();
        obtenerRankings();      
    }
}

File: Práctica 1/S1/S1R/lib/Proyecto.rb
#encoding: UTF-8
require_relative "Cliente.rb"

module Ruby
	class Proyecto
  
    @cliente=Cliente.new
  
   
    def self.corrercarreras
      for i in @cliente.carreras
        i.correr
      end
    end
    
    def self.obtenerranking
      for i in @cliente.carreras
        i.mostrarranking
      end
    end
  
    
		def self.main
      @cliente.gestionar
      corrercarreras
      obtenerranking
    end
    
	end
  Proyecto.main
end
